.github/workflows/run_test_suite.yaml

```diff
@@ -1,35 +1,66 @@
-on: [push, pull_request]
+name: Tests and Checks
 
-name: Run test suite
+on:
+  push:
+    branches: [ main ]
+
+  pull_request:
+    branches: [ main ]
```

Comment:
Do we want to limit this to pull requests against main?

Reply:
will make this *